client/src/featureform/serving.py

```diff
@@ -387,10 +387,41 @@ def get_input_df(self, source_name, source_variant):
             else:
                 raise ValueError(f"Unsupported file format for {file_path}")
             return df
+        elif (
+            self.db.is_transformation(source_name, source_variant)
+            == SourceType.DIRECTORY
+        ):
+            source = self.db.get_source_variant(source_name, source_variant)
+            directory = source["definition"]
+            return self.read_directory(directory)
+
         else:
             df = self.process_transformation(source_name, source_variant)
         return df
 
+    def read_directory(self, directory):
+        if not os.path.isdir(directory):
+            raise Exception(f"Path {directory} is not a directory")
+
+        def absolute_file_paths(directory):
```

Comment:
if we're using this function in multiple places we can add this somewhere else instead of copying it over to the cache as well. file_utils.py sounds like it could work. or even a utils.

Reply:
Updated